browser/ui/views/toolbar/brave_toolbar_view.cc

```diff
@@ -169,19 +179,89 @@ void BraveToolbarView::Layout() {
+  // but consider the minimum for the location bar
+  const int location_bar_min_width = location_bar_->GetMinimumSize().width();
   const int browser_actions_width = browser_actions_->GetWidthForMaxWidth(
-      available_width - location_bar_->GetMinimumSize().width());
+      available_width - location_bar_min_width);
   available_width -= browser_actions_width;
-  const int location_bar_width = available_width;
 
+  // Allow the option of the LocationBar having horizontal margin.
+  // With this option, we support a dynamic percentage of margin, with
+  // a maximum width.
+  const bool restrict_location_bar_width = !location_bar_is_wide_.GetValue();
+  const int location_bar_max_width = restrict_location_bar_width
+                                      ? 1080
+                                      : available_width;
+  const auto toolbar_width = width();
+  int location_bar_width = available_width;
+  int location_bar_margin_h = 0;
+  int location_bar_center_offset = 0;
+  if (restrict_location_bar_width) {
+    double location_bar_margin_h_pc = 0.07;
+    if (toolbar_width < 700)
+      location_bar_margin_h_pc = 0;
+    else if (toolbar_width < 850)
+      location_bar_margin_h_pc = 0.03;
+    else if (toolbar_width < 1000)
+      location_bar_margin_h_pc = 0.05;
+    // Apply the target margin, adjusting for min and max width of LocationBar
+    // Make sure any margin doesn't shrink the LocationBar beyond minimum width
+    if (available_width > location_bar_min_width) {
+      int location_bar_max_margin_h = (available_width - location_bar_min_width) /
+                                      2;
+      location_bar_margin_h = std::min(
+                                  (int)(toolbar_width * location_bar_margin_h_pc),
+                                  location_bar_max_margin_h);
+      location_bar_width = available_width - (location_bar_margin_h * 2);
+      // Allow the margin to expand so LocationBar is restrained to max width
+      if (location_bar_width > location_bar_max_width) {
+        location_bar_margin_h += (location_bar_width - location_bar_max_width) /
+                                  2;
+        location_bar_width = location_bar_max_width;
+      }
+    }
+    // Center LocationBar as much as possible within Toolbar
+    const int location_bar_toolbar_center_point =
+                                        next_element_x +
+                                        location_bar_margin_h +
+                                        (location_bar_width / 2);
+    // Calculate offset - positive for move left and negative for move right
+    location_bar_center_offset = location_bar_toolbar_center_point -
+                                                            (toolbar_width / 2);
+    // Can't shim more than we have space for, so restrict to margin size
+    // or in the case of moving-right, 25% of the space since we want to avoid
+    // touching browser actions where possible
```

Comment:
I can't understand this comment about why right-side moving is restricted to 25% of margin.
There are always gap between location bar and browser action area because browser actions has it's internal padding.

Reply:
This is to prevent the gap being larger on the left side than the right side, reducing the extra margin (the gap that does not include the left padding on the browser actions area). Whilst we aim to center the LocationBar as much as possible, a more important priority is to not be in the situation where there is *some* gap on the left, and *no gap* on the right, due to the proximity of the brave actions bar and the browser actions bar.

Of course, when there is no extra margin (either user-configurable or because the window is too narrow), then we are ok with no extra gap on either side.

Without shim-right restriction (moves right, towards center):
![image](https://user-images.githubusercontent.com/741836/45639357-a30f8780-ba64-11e8-8d33-6a6a93d4eeb8.png)

With 100% shim-right restriction (not allowed a smaller gap on right than left):
![image](https://user-images.githubusercontent.com/741836/45639396-c4707380-ba64-11e8-97b3-fd3492b37ce0.png)

With 25% shim-right restriction (can move right towards center, but not more than 25% of the right-margin):
![image](https://user-images.githubusercontent.com/741836/45639729-993a5400-ba65-11e8-8806-d81c62bb5097.png)

(Apologies for the discrepancy of the bookmark button position, some screenshots were taken before this moved!)

Reply:
Thanks for explanation! I thought location bar view should be center on toolbar always.